### ISSUE TITLE
Unable to specify correct filter to query and return all Project, Episode, and Shot fields together

### ISSUE THREAD

**Issue by omac777** [shotgunsoftware/python-api#273]
Using the following variables:
```
project_name = 'SuperStickman'
episode_name = 'Stic-101'
```

I want to extract ALL the Project, Episode and Shot fields together.

I tried these two different ways, but they don't display Project, Episode and Shot fields.
```
sg_shots = sg.find("Shot", [['project.Project.name', 'is', project_name],['episode.Episode.name', 'is', episode_name]], ['code', 'sg_sequence.Sequence.sg_status_list'])
```

```
sg_shots = sg.find("Shot", [['project.Project.name', 'is', project_name]], ['code', 'sg_sequence.Sequence.sg_status_list'])
```

How do I return a query result that returns ALL the Project, Episode and Shot fields together?

Thank you in advance.

**Comment by monkeydev** [comment]
Hello,
You just need to add ‘episode.Episode.name’ and ‘project.Project.name’ to the list of fields to retrieve, if I understood what you’re after.
Something like: ['code', ‘sg_sequence.Sequence.sg_status_list’,  ‘episode.Episode.name’, ‘project.Project.name’ ]
Hope it helps.

**Comment by omac777** [comment]
I tried this as well:
```
sg_shots = sg.find("Shot", [['project.Project.name', 'is', project_name],['episode.Episode.name', 'is', episode_name]], ['code', 'sg_sequence.Sequence.sg_status_list',  'episode.Episode.name', 'project.Project.name' ] )
```

But it returns with an error:
```
shotgun_api3.shotgun.Fault: API read() Shot.episode.Episode.name doesn't exist:
{"path"=>"episode.Episode.name", "relation"=>"is", "values"=>["Stic-101"]}
```

**Comment by monkeydev** [comment]
I guess your Shot was not linked to an Episode.
Just add “episode” to the list of fields to return to double check: if empty then the Shot is not linked to an episode.
If the episode is set on the Sequence, you can get it with `sg_sequence.Sequence.episode`, and its field values with  `sg_sequence.Sequence.episode.Episode.code`
Check your SG schema for exact field values.
Cheers.

**Comment by omac777** [comment]
Much appreciated.  Thanks you will try those.

**Comment by omac777** [comment]
BINGO:
```
sg_shots = sg.find("Shot", [['project.Project.name', 'is', project_name]], ['code', 'sg_sequence.Sequence.sg_status_list',  'sg_sequence.Sequence.episode', 'sg_sequence.Sequence.episode.Episode.code' ] )
```
Returns:

{'type': 'Shot', 'id': 8345, 'code': 'blahcodes1', 'sg_sequence.Sequence.sg_status_list': 'blahstatus2', 'sg_sequence.Sequence.episode': {'id': 400, 'name': 'Stic-101', 'type': 'Episode'}, 'sg_sequence.Sequence.episode.Episode.code': 'Stic-101'}

Thank you.  We've got what we need with this. :)

You're a super-hero monkeydev.

**Comment by omac777** [comment]
How do I fetch the task name within this query as well?
```
sg_shots = sg.find("Shot", [['project.Project.name', 'is', project_name]], ['code', 'sg_sequence.Sequence.sg_status_list',  'sg_sequence.Sequence.episode', 'sg_sequence.Sequence.episode.Episode.code', 'task.Task.name' ] )
```
I tried to fetch it but it's empty as well. Not linked as you mentioned?

**Comment by blerrgh** [comment]
@josh-mission I don't believe that is possible for multi-entity fields.  see: https://community.shotgridsoftware.com/t/using-linked-fields-aka-bubbled-fields-in-api-requests/8697
